core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -196,27 +206,46 @@ export class TranslationModalComponent {
   ngOnInit(): void {
     this.activeLanguageCode =
       this.translationLanguageService.getActiveLanguageCode();
-    // We need to set the context here so that the rte fetches
-    // images for the given ENTITY_TYPE and targetId.
-    this.contextService.setCustomEntityContext(
-      AppConstants.ENTITY_TYPE.EXPLORATION,
-      this.opportunity.id
-    );
-    this.subheading = this.opportunity.subheading;
-    this.heading = this.opportunity.heading;
+    this.subheading = this.opportunity
+      ? this.opportunity.subheading
+      : this.modifyTranslationOpportunity.subheading;
+    this.heading = this.opportunity
+      ? this.opportunity.heading
+      : this.modifyTranslationOpportunity.heading;
     this.contextService.setImageSaveDestinationToLocalStorage();
     this.languageDescription =
       this.translationLanguageService.getActiveLanguageDescription();
-    this.translateTextService.init(
-      this.opportunity.id,
-      this.translationLanguageService.getActiveLanguageCode(),
-      () => {
-        const translatableItem = this.translateTextService.getTextToTranslate();
-        this.updateActiveState(translatableItem);
-        ({more: this.moreAvailable} = translatableItem);
-        this.loadingData = false;
-      }
-    );
+
+    if (!this.modifyTranslationOpportunity) {
+      // We need to set the context here so that the rte fetches
+      // images for the given ENTITY_TYPE and targetId.
+      this.contextService.setCustomEntityContext(
+        AppConstants.ENTITY_TYPE.EXPLORATION,
+        this.opportunity.id
+      );
+
+      this.translateTextService.init(
+        this.opportunity.id,
+        this.translationLanguageService.getActiveLanguageCode(),
+        () => {
+          const translatableItem =
+            this.translateTextService.getTextToTranslate();
+          this.updateActiveState(translatableItem);
+          ({more: this.moreAvailable} = translatableItem);
+          this.loadingData = false;
+        }
+      );
+    } else {
+      this.textToTranslate = this.modifyTranslationOpportunity.textToTranslate;
+      this.activeContentType =
```

Comment:
Why can't we use `getFormattedContentType()` to do this?

Reply:
It was just because that function needed the interaction ID as well that we don't need elsewhere, fixed it